app/src/main/java/com/appdev/eateryblueandroid/util/RecentSearchesUtils.kt

```diff
@@ -0,0 +1,76 @@
+package com.appdev.eateryblueandroid.util
+
+import com.appdev.eateryblueandroid.util.Constants.userPreferencesStore
+import com.codelab.android.datastore.AccountProto
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+
+
+/**
+ * A list of ints corresponding to the recent ID's that the user has
+ * searched and clicked on.
+ *
+ * Use saveRecentSearch(search) to add recent searches to the list.
+ *
+ * Index 0 corresponds to the newest search, index size-1 is the oldest search.
+ */
+val recentSearches : List<Int> = mutableListOf()
```

Comment:
You seem to cast the list a lot. Instead, do you want to just set the type of this to MutableList?